### PR TITLE
(MODULES-2520) Add link to Types document

### PR DESCRIPTION
**Pull request by jpogran**
Add a reference and link to the types document that is automatically
generated when building the dsc_* types.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,12 +15,13 @@
   * [Adding or Removing Windows Features](#adding-or-removing-windows-features)
   * [Website Installation Example](#website-installation-example)
   * [Optionally Configure the DSC LCM RefreshMode](#optionally-configure-the-dsc-lcm-refreshmode)
-5. [Limitations](#limitations)
+5. [Reference - An under-the-hood peek at what the module is doing and how](#reference)
```
Comment (jpogran) [inline README.md]: Copied this to aline with other module documentation, but I don't know that it's valid given this section does not explain 'what the module is doing and how'

Comment (ferventcoder) [inline README.md]: Looks great.

File: README.md [hunk 2]
```diff
@@ -297,6 +298,12 @@ dsc::lcm_config {'disable_lcm':
 }
 ~~~
 
+## Reference
+
+### Types
+
+A comprehensive listing of all types included in the dsc module is available in the [types document](https://github.com/puppetlabs/puppetlabs-dsc/blob/master/types.md).
+
```
Comment (ferventcoder) [inline README.md]: The docs review will overhaul this if they don't find it valid.

File: README.md (hunk 1 above)
Comment (cowofevil) [inline README.md]: This looks weird. I think it is more intuitive to have the TOC just have "Reference" without the extra annotation.

Comment (ferventcoder) [inline README.md]: It's kind of a starter. I imagine a docs review will start to fill out this section some more.

**Pull request merged**